cmd/otel-allocator/watcher/main.go

```diff
@@ -33,6 +33,13 @@ const (
 	EventSourcePrometheusCR
 )
 
+var (
+	EventSourceToString = map[EventSource]string{
+		EventSourceConfigMap:    "EventSourceConfigMap",
+		EventSourcePrometheusCR: "EventSourcePrometheusCR",
+	}
+)
+
```

Comment:
```suggestion
	eventSourceToString = map[EventSource]string{
		EventSourceConfigMap:    "EventSourceConfigMap",
		EventSourcePrometheusCR: "EventSourcePrometheusCR",
	}
)

func (e EventSource) String() string {
	return eventSourceToString[e]
}
```

Avoid exporting the mutable map and instead implement `Stringer` for `EventSource`.